addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -348,6 +348,58 @@ var FieldChar = InputField.extend(TranslatableFieldMixin, {
     },
 });
 
+
+var LinkButton = FieldChar.extend({
+    events: {
+        'click': '_onClick'
+    },
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     * @private
+     */
+    _render: function () {
+        this._super.apply(this, arguments);
+        if (this.value) {
+            this._renderLinkButton();
+        }
```

Comment:
You may want to force the rendering mode to readonly... If so, you can explicitely call this._renderReadonly() instead of this._super() here.